File: src/main/java/com/example/sneakerx/entities/AttributeValue.java
package com.example.sneakerx.entities;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "attribute_values")
public class AttributeValue {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer valueId;

    @ManyToOne
    @JoinColumn(name = "attribute_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private ProductAttribute attribute;

    private String value;
    private String imageUrl;
}

File: lib/src/models/attribute_value.dart
class AttributeValue {
  final int valueId;
  final int attributeId;
  final String value;
  final String? imageUrl;

  AttributeValue({
    required this.valueId,
    required this.value,
    required this.attributeId,
    this.imageUrl,
  });

  factory AttributeValue.fromJson(Map<String, dynamic> json) {
    return AttributeValue(
      valueId: json['valueId'],
      attributeId: json['attributeId'],
      value: json['value'],
      imageUrl: json['imageUrl'] as String?,
    );
  }


}